subsys/net/lib/nrf_cloud/coap/include/nrf_cloud_coap_transport.h

```diff
@@ -33,11 +33,13 @@ enum coap_content_format {
 #endif
 
 struct nrf_cloud_coap_client {
+	struct k_mutex mutex;
```

Comment:
the lighter-weight `k_spinlock` could be used here instead.

Reply:
This will make the calling thread non-preemtible during the acquisition of the spinlock, won't it? I don't personally feel that is desirable -- the overhead seems worth allowing users to call these APIs without disabling preemption

Reply:
i don't see why that would matter in this use case; ensuring data integrity when changing a few variables.
pete can decide.

Reply:
We do call some socket APIs in, for example, `nrf_cloud_coap_transport_disconnect`, and we also call stuff like `nrfc_dtls_session_save` for instance in `nrf_cloud_coap_transport_pause`

Reply:
spinlocks disable interrupts; I don't want to use them for this